src/sentry/snuba/metrics/naming_layer/mri.py

```diff
@@ -298,11 +307,46 @@ def parse_mri(mri_string: str | None) -> ParsedMRI | None:
     return ParsedMRI(**match.groupdict())
 
 
+def parse_mri_lenient(mri_string: str | None) -> ParsedMRI | None:
+    """
+    Parse a mri string in a lenient way, meaning that the goal of the parsing is just
+    to find the common structure of the mri which is split by a series of known delimiters ':', '/', '@'.
+    """
+    if mri_string is None:
+        return None
+
+    mri_delimiters = [":", "/", "@"]
+    mri_components = []
+    start = 0
+
+    # Find the MRI delimiters one after the other.
+    for i in range(len(mri_string)):
+        if mri_delimiters and mri_string[i] in mri_delimiters[0]:
+            mri_components.append(mri_string[start:i])
+            mri_delimiters.pop(0)
+            start = i + 1
+
+    # Append the substring after the last delimiter. If we have no string after the last delimiter, we will still return
+    # ''.
+    mri_components.append(mri_string[start:])
+
+    # In case we do not find the exact number of components that we expect in the MRI, we will return `None`.
+    if len(mri_components) != 4:
```

Comment:
Did we test this case -- a case where we might not have 4 components due to a missing delimiter?

Reply:
Oh yeah, I added the edge case of nothing after the `@` but not in the middle. Will add now.

Reply:
Nvm, I checked again. I did add that test, it's here: https://github.com/getsentry/sentry/blob/6a589b1eafc01321960ba4190cdb131d20fd3276/tests/sentry/snuba/metrics/test_naming_layer.py#L165-L165

Reply:
And also here: https://github.com/getsentry/sentry/blob/6a589b1eafc01321960ba4190cdb131d20fd3276/tests/sentry/snuba/metrics/test_naming_layer.py#L164-L164

Reply:
ok thanks for checking!